packages/dev/core/src/Meshes/abstractMesh.ts

```diff
@@ -536,10 +542,41 @@ export abstract class AbstractMesh extends TransformNode implements IDisposable,
      */
     public alphaIndex = Number.MAX_VALUE;
 
+    /**
+     * If set to true, a mesh will only be visible only if its parent(s) are also visible (default is false)
+     */
+    public get inheritedVisibility(): boolean {
+        return this._internalAbstractMeshDataInfo._inheritedVisibility;
+    }
+
+    public set inheritedVisibility(value: boolean) {
+        this._internalAbstractMeshDataInfo._inheritedVisibility = value;
+    }
+
+    private _isVisible = true;
     /**
      * Gets or sets a boolean indicating if the mesh is visible (renderable). Default is true
      */
-    public isVisible = true;
+    public get isVisible(): boolean {
+        if (!this._isVisible || !this.inheritedVisibility || !this._parentNode) {
+            return this._isVisible;
+        }
+        if (this._isVisible) {
+            let parent: Nullable<Node> = this._parentNode;
+            while (parent) {
+                const parentVisible = (parent as AbstractMesh).isVisible;
+                if (typeof parentVisible !== "undefined") {
+                    return this._isVisible && parentVisible;

```

Comment:
This checks _isVisible but it's in a block that you'll only reach if _isVisible is true - can this just be return parentVisible?

Reply:
makes sense :-)